phys/module_microphysics_driver.F

```diff
@@ -2566,6 +2567,58 @@ SUBROUTINE microphysics_driver(                                          &
                CALL wrf_error_fatal ( 'arguments not present for calling wdm7')
              ENDIF
 
+        CASE (UDMSCHEME)
+             CALL wrf_debug ( 100 , 'microphysics_driver: calling udm' )
+             IF ( PRESENT( QV_CURR ) .AND. PRESENT ( QC_CURR ) .AND.  &
+                  PRESENT( QR_CURR ) .AND. PRESENT ( QI_CURR ) .AND.  &
+                  PRESENT( QS_CURR ) .AND. PRESENT ( QG_CURR ) .AND.  &
+                  PRESENT( QH_CURR ) .AND.                            &
+                  PRESENT( QNN_CURR ) .AND. PRESENT ( QNC_CURR ) .AND. &
+                  PRESENT( QNR_CURR ).AND.                            &
+                  PRESENT( RAINNC  ) .AND. PRESENT ( RAINNCV )  ) THEN
```

Comment:
Do we still check all these variables' presence if we only have one dynamic core?

Reply:
I think this is part of a bigger question, but if they are optional in the driver we need these.